app/src/main/java/com/cp/tasky/auth/shared/presentation/AuthViewModel.kt

```diff
@@ -0,0 +1,22 @@
+package com.cp.tasky.auth.shared.presentation
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.cp.tasky.auth.shared.domain.AuthRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class AuthViewModel @Inject constructor(
+    private val authRepository: AuthRepository,
+) : ViewModel() {
+
+    var isAuthenticatedUser by mutableStateOf(false)
+        private set
+
+    fun isAuthenticated() {
+        isAuthenticatedUser = authRepository.isAuthenticatedUser()
+    }
```

Comment:
To implement my above suggestion, you'd need an additional boolean state here when you're currently checking authentication